api/src/main/java/org/openmrs/module/pihcore/task/PihRemovePatientsFromMCOEQueue.java

```diff
@@ -0,0 +1,106 @@
+package org.openmrs.module.pihcore.task;
+
+import org.apache.commons.lang.time.StopWatch;
+import org.openmrs.Concept;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.queue.api.QueueServicesWrapper;
+import org.openmrs.module.queue.api.search.QueueEntrySearchCriteria;
+import org.openmrs.module.queue.api.search.QueueSearchCriteria;
+import org.openmrs.module.queue.model.Queue;
+import org.openmrs.module.queue.model.QueueEntry;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Duration;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+
+public class PihRemovePatientsFromMCOEQueue implements Runnable {
+
+    private static final Long QUEUE_TIMEOUT_MINUTES = 720L;
+    private static final String MCOE_TRIAGE_QUEUE_CONCEPT_CODE_PIH = "14976";
+
+    private final Logger log = LoggerFactory.getLogger(getClass());
+
+    private static boolean isExecuting = false;
+
+    @Override
+    public void run() {
+
+        if (isExecuting) {
+            log.info("Not executing " + getClass() + " because another instance is already executing");
+            return;
+        }
+
+        isExecuting = true;
+
+        try {
+            log.info("Executing " + getClass());
+            StopWatch stopWatch = new StopWatch();
+            stopWatch.start();
+
+            Queue queue = getQueue();
+            if (queue == null) return;
+
+            int removedPatientsCount = removeTimedOutPatients(queue);
+
+            stopWatch.stop();
+            log.info("Execution completed in {} - {} patients removed", stopWatch, removedPatientsCount);
+        } catch (Exception e) {
+            log.error("Error executing " + getClass(), e);
+        } finally {
+            isExecuting = false;
+        }
+    }
+
+    private Queue getQueue() throws Exception {
```

Comment:
Isn't this unnecessarily complex?  Why not just retrieve queue by uuid?  The MCOE Triage Queue uuid is a fixed value - 3113f164-68f0-11ee-ab8d-0242ac120002